### PR TITLE
Login passport

### PR DESCRIPTION
**Pull request by DarylTew**
multiple sessions now available using passport

### CODE REVIEW

**Review by AyushChatto — APPROVED** [review]
LGTM - this is absolutely beastly :D

**Review by chelchia — APPROVED** [review]
looks great!!

File: config/passport.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        var pool = req.app.get('pool');
+
+        // begin transaction, don't commit unless all makes it through
+        pool.query('BEGIN', function(err) {
+            if(err) {
+                console.log('begin error');
+                return done(err);
+            }
+            // find a user whose username is the same as the forms username
+            // we are checking to see if the user trying to login already exists
```
Comment (adamwth) [inline config/passport.js:90]: user trying to sign up*

File: config/passport.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            pool.query('INSERT INTO accounts (username, password) VALUES ($1, $2);', [username, password], function(err, user) {
+                // check to see if theres already a user with that username
+                if (!user) {
```
Comment (adamwth) [inline config/passport.js:93]: shouldn't you be checking the `err` instead?

**Review by adamwth — APPROVED** [review]
lgtm. if tested on local and doesn't break anything, just merge.

File: config/passport.js (hunk 2 above)
Comment (DarylTew) [reply config/passport.js:93]: it's either i return done(err) which shows me a page of duplicate key error or I return the user an flash error message that is more useful haha

**Pull request merged**